### PR TITLE
Development

### PR DESCRIPTION
**Pull request by DariusDangi**

### CODE REVIEW

File: ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_NavigateToSFBooks.xaml
Comment (FlorinBarsan) [inline ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_NavigateToSFBooks.xaml:1]: verify selectors from this workflow:
attach window unified application target selector browser URL contains a GitHub address
delete all selectors that are not checked in targeting method
empty spaces and sketchy characters should be replaced with wildcards

File: ApplicationLayer/Chrome/eMAG/Chrome_eMAG_NavigateToSFBooks.xaml
Comment (FlorinBarsan) [inline ApplicationLayer/Chrome/eMAG/Chrome_eMAG_NavigateToSFBooks.xaml:1]: verify selectors from this workflow:
delete all selectors that are not checked in targeting method
empty spaces and sketchy characters ([], &amp) should be replaced with wildcards

File: ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_ApplyFilters.xaml
Comment (FlorinBarsan) [inline ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_ApplyFilters.xaml:1]: select ascending price - anchor should have also a strict/fuzzy selector available, we shouldn't rely only on computer vision
- delete targeting method selectors that are not used
- select year of publish : what's the target there? I saw 2 anchors  in which one has as text the 2023 (which should be the target). Plus, idx used

File: ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_ExtractBooks.xaml
Comment (FlorinBarsan) [inline ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_ExtractBooks.xaml:1]: - "Do" sequence is redundant
- delete targeting method selectors that are not used

File: ApplicationLayer/Chrome/eMAG/Chrome_eMAG_ApplyFilters.xaml
Comment (FlorinBarsan) [inline ApplicationLayer/Chrome/eMAG/Chrome_eMAG_ApplyFilters.xaml:1]: - delete all selectors that are not checked in targeting method
- empty spaces and sketchy characters should be replaced with wildcards
- default retry scope name
- 
image

File: ApplicationLayer/Chrome/eMAG/Chrome_eMAG_ExtractBooks.xaml
Comment (FlorinBarsan) [inline ApplicationLayer/Chrome/eMAG/Chrome_eMAG_ExtractBooks.xaml:1]: - "Do" sequence is redundant
- delete targeting method selectors that are not used

File: ApplicationLayer/Chrome/eMAG/Chrome_eMAG_GetBookDetails.xaml
Comment (FlorinBarsan) [inline ApplicationLayer/Chrome/eMAG/Chrome_eMAG_GetBookDetails.xaml:1]: - delete targeting method selectors that are not used
- when creating unique reference, please store the format in config with something like "1_<datePlaceholder>_<TitleLength>", then on the actual code use Repalce method to replace the placeholders with the necessary info. This allow you to not have any hardcoded parameters

File: DataLayer/UploadQueueItems.xaml
Comment (FlorinBarsan) [inline DataLayer/UploadQueueItems.xaml:1]: you are using orchestrator service in this workflow to upload items, so I'd move this xaml file to ServiceLayer

File: Framework/CloseAllApplications.xaml
Comment (FlorinBarsan) [inline Framework/CloseAllApplications.xaml:1]: redundant sequences

**Review by FlorinBarsan — CHANGES_REQUESTED** [review]
Please be more careful when working with selectors and clean what isn't needed. Overall nice implementation but pay more attention to small details!

File: ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_NavigateToSFBooks.xaml
Comment (DariusDangi) [reply ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_NavigateToSFBooks.xaml:1]: Removed unsued selectors that are not checked in targeting methods, rectified selectors that had spaces/sketchy characters and removed the GitHub url from Use Browser activity

File: ApplicationLayer/Chrome/eMAG/Chrome_eMAG_NavigateToSFBooks.xaml
Comment (DariusDangi) [reply ApplicationLayer/Chrome/eMAG/Chrome_eMAG_NavigateToSFBooks.xaml:1]: Removed unused selectors by targeting methods and modified the ones that are used by using wildcards.

File: ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_ApplyFilters.xaml
Comment (DariusDangi) [reply ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_ApplyFilters.xaml:1]: -Fixed the selector for ascending price. Tbh, i didn't even realize i had only computer vision on that activity.
-Removed unused selectors.
-The target is the checked checkbox. Those two anchors are for the verification of execution. The checked checkbox has a few attributes and that's why i chose to use two anchors.

File: ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_ExtractBooks.xaml
Comment (DariusDangi) [reply ApplicationLayer/Chrome/Carturesti/Chrome_Carturesti_ExtractBooks.xaml:1]: Removed unused selectors and the redundant "Do" sequence

File: ApplicationLayer/Chrome/eMAG/Chrome_eMAG_ApplyFilters.xaml
Comment (DariusDangi) [reply ApplicationLayer/Chrome/eMAG/Chrome_eMAG_ApplyFilters.xaml:1]: -Deleted all selectors that are not used
-Modified used selectors to replace spaces and sketchy characters with wildcards.
-Renamed activity that has default name

File: ApplicationLayer/Chrome/eMAG/Chrome_eMAG_ExtractBooks.xaml
Comment (DariusDangi) [reply ApplicationLayer/Chrome/eMAG/Chrome_eMAG_ExtractBooks.xaml:1]: Removed unused selectors and redundant "Do" sequence

File: ApplicationLayer/Chrome/eMAG/Chrome_eMAG_GetBookDetails.xaml
Comment (DariusDangi) [reply ApplicationLayer/Chrome/eMAG/Chrome_eMAG_GetBookDetails.xaml:1]: Created fields in config that will store a code (0 or 1) that will be used when building the unique reference.
Removed unused selectors.

File: DataLayer/UploadQueueItems.xaml
Comment (DariusDangi) [reply DataLayer/UploadQueueItems.xaml:1]: Moved UploadQueueItems workflow from DataLayer to ServiceLayer.

File: Framework/CloseAllApplications.xaml
Comment (DariusDangi) [reply Framework/CloseAllApplications.xaml:1]: Removed redundant sequences.

**Review by FlorinBarsan — APPROVED** [review]
Good job!

**Pull request merged**